[refactor] Adding FoldConvBatchNorm2dHelper

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #32379 [WIP][quant][graphmode] FoldConvBatchNorm2d support shared ClassTypes
* #32543 [WIP][quant][graphmode] ConvBn Folding works with bias=None
* #32380 [quant][graphmode][refactor] Better API for fold_convbn
* **#32374 [refactor] Adding FoldConvBatchNorm2dHelper**
* #32303 [quant][graphmode] Call _jit_pass_dedup_module_ueses in quantize_script
* #32302 [quant] Re-enable fold_convbn in quantize_script
* #32206 [quant] Re-enable test_nested that has different qconfig for shared ClassType
* #32205 [quant][graphmode] Support quantizing shared ClassType with different qconfigs

Summary:
Moving all fold conv bn code to a class to prepare for making
it work with shared ClassType

Test Plan:
compiles

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

Differential Revision: [D19508032](https://our.internmc.facebook.com/intern/diff/D19508032)

jerryzh168 committed:
[refactor] Adding FoldConvBatchNorm2dHelper

Summary:
Moving all fold conv bn code to a class to prepare for making
it work with shared ClassType

Test Plan:
compiles

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[refactor] Adding FoldConvBatchNorm2dHelper"


Summary:
Moving all fold conv bn code to a class to prepare for making
it work with shared ClassType

Test Plan:
compiles

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

kostmo commented:
## :pill: CircleCI build failures summary and remediations
As of commit d8f2d2f3:


**None of the build failures appear to be your fault**.

* **1/1** recognized as flaky :snowflake:
    * Re-run these jobs?


## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=d8f2d2f320f5a29a78979746d8df8f6517db9a52).

### :snowflake: 1 failure recognized as flaky
The following build failures have been detected as flaky and **may not be your fault**:
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/4294256) caffe2_onnx_py3_6_clang7_ubuntu16_04_test (1/1)
**Step:** "Test" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=87964951) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=87964951)) :snowflake:
<details>
<summary>
<code>Jan 17 23:23:56 test/onnx/test_utility_funs.py::TestUtilityFuns::test_constant_fold_reshape Fatal Python error: Segmentation fault</code>
</summary>

```
Jan 17 23:23:56 test/onnx/test_utility_funs.py::TestUtilityFuns_opset11::test_constant_fold_transpose_matmul PASSED [ 99%] 
Jan 17 23:23:56 test/onnx/test_utility_funs.py::TestUtilityFuns_opset11::test_constant_fold_unsqueeze PASSED [ 99%] 
Jan 17 23:23:56 test/onnx/test_utility_funs.py::TestUtilityFuns_opset11::test_error_on_data_parallel PASSED [ 99%] 
Jan 17 23:23:56 test/onnx/test_utility_funs.py::TestUtilityFuns_opset11::test_is_in_onnx_export PASSED [ 99%] 
Jan 17 23:23:56 test/onnx/test_utility_funs.py::TestUtilityFuns_opset11::test_strip_doc_string PASSED [ 99%] 
Jan 17 23:23:56 test/onnx/test_utility_funs.py::TestUtilityFuns_opset11::test_validate_dynamic_axes_invalid_input_output_name PASSED [ 99%] 
Jan 17 23:23:56 test/onnx/test_utility_funs.py::TestUtilityFuns::test_constant_fold_concat PASSED [ 99%] 
Jan 17 23:23:56 test/onnx/test_utility_funs.py::TestUtilityFuns::test_constant_fold_div PASSED [ 99%] 
Jan 17 23:23:56 test/onnx/test_utility_funs.py::TestUtilityFuns::test_constant_fold_lstm PASSED [ 99%] 
Jan 17 23:23:56 test/onnx/test_utility_funs.py::TestUtilityFuns::test_constant_fold_mul PASSED [ 99%] 
Jan 17 23:23:56 test/onnx/test_utility_funs.py::TestUtilityFuns::test_constant_fold_reshape Fatal Python error: Segmentation fault 
Jan 17 23:23:56  
Jan 17 23:23:56 Current thread 0x00007fdd85d4f700 (most recent call first): 
Jan 17 23:23:56   File "/usr/local/lib/python3.6/dist-packages/_pytest/_io/saferepr.py", line 43 in repr_instance 
Jan 17 23:23:56   File "/usr/lib/python3.6/reprlib.py", line 65 in repr1 
Jan 17 23:23:56   File "/usr/lib/python3.6/reprlib.py", line 55 in repr 
Jan 17 23:23:56   File "/usr/local/lib/python3.6/dist-packages/_pytest/_io/saferepr.py", line 36 in repr 
Jan 17 23:23:56   File "/usr/local/lib/python3.6/dist-packages/_pytest/_io/saferepr.py", line 67 in saferepr 
Jan 17 23:23:56   File "/usr/local/lib/python3.6/dist-packages/_pytest/_code/code.py", line 655 in repr_args 
Jan 17 23:23:56   File "/usr/local/lib/python3.6/dist-packages/_pytest/_code/code.py", line 736 in repr_traceback_entry 
Jan 17 23:23:56   File "/usr/local/lib/python3.6/dist-packages/_pytest/_code/code.py", line 777 in repr_traceback 
```

</details>



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

jerryzh168 committed:
Update on "[refactor] Adding FoldConvBatchNorm2dHelper"


Summary:
Moving all fold conv bn code to a class to prepare for making
it work with shared ClassType

Test Plan:
compiles

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[refactor] Adding FoldConvBatchNorm2dHelper"


Summary:
Moving all fold conv bn code to a class to prepare for making
it work with shared ClassType

Test Plan:
compiles

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

ZolotukhinM reviewed (approved):
👍

jerryzh168 committed:
Update on "[refactor] Adding FoldConvBatchNorm2dHelper"


Summary:
Moving all fold conv bn code to a class to prepare for making
it work with shared ClassType

Test Plan:
compiles

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[refactor] Adding FoldConvBatchNorm2dHelper"


Summary:
Moving all fold conv bn code to a class to prepare for making
it work with shared ClassType

Test Plan:
compiles

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[refactor] Adding FoldConvBatchNorm2dHelper"


Summary:
Moving all fold conv bn code to a class to prepare for making
it work with shared ClassType

Test Plan:
compiles

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[refactor] Adding FoldConvBatchNorm2dHelper"


Summary:
Moving all fold conv bn code to a class to prepare for making
it work with shared ClassType

Test Plan:
compiles

Reviewers:
mvz

Subscribers:

Tasks:

Tags:

Differential Revision: [D19508032](https://our.internmc.facebook.com/intern/diff/D19508032)

[ghstack-poisoned]
